### ISSUE TITLE
GPTBigCode models generate noise results

### ISSUE THREAD

**Issue by anxietymonger** [vllm-project/vllm#675]
https://github.com/vllm-project/vllm/blob/aa84c92ef636e689b506b9842c712e5c615cc73a/vllm/config.py#L101

Adding conditions for falcon models seems break inference for GPTBigCode models. It worked well on commit df5dd3c68ec7007e4b3a27907f4d954f1c7d511d, but broke on commit aa84c92ef636e689b506b9842c712e5c615cc73a.

I have tried WizardCoder 15B, if I change the condition
```python
if (getattr(self.hf_config, "multi_query", False) and
            (self.hf_config.model_type == "falcon" and
             not getattr(self.hf_config, "new_decoder_architecture", False))):
```
into
```python
if getattr(self.hf_config, "multi_query", False):
```
, it works well.

I'm not sure whether this is a bug, or I just need some extra configuration?

**Comment by anxietymonger** [comment]
Maybe it should be an `or`?

**Comment by zhuohan123** [comment]
Fixed by #676. Thank you for reporting this!
